spec/models/course/publishable_spec.rb

```diff
@@ -9,15 +11,33 @@
     context 'with enrichment' do
       let(:enrichment) { build(:course_enrichment, :subsequent_draft, created_at: 1.day.ago) }
       let(:course) {
-        create(:course, enrichments: [enrichment])
+        create(:course, enrichments: [enrichment], site_statuses: [site_status])
       }
 
       its(:publishable?) { should be_truthy }
     end
 
     context 'with no enrichment' do
       let(:course) {
-        create(:course)
+        create(:course, site_statuses: [site_status])
+      }
+
+      its(:publishable?) { should be_falsey }
+
+      describe 'course errors' do
+        subject do
+          course.publishable?
+          course.errors
+        end
+
+        it { should_not be_empty }
+      end
+    end
+
+    context 'with no sites' do
+      let(:enrichment) { build(:course_enrichment, :subsequent_draft, created_at: 1.day.ago) }
+      let(:course) {
+        create(:course, enrichments: [enrichment])
```

Comment:
Should we explicitly set `site_statuses` as empty here?

```suggestion
        create(:course, site_statuses: [], enrichments: [enrichment])
```